### PR TITLE
Jupiter: Required fields error validations

### PR DESCRIPTION
**Pull request by anuritjain**
https://jira.rax.io/browse/JUPITER-9.   //Required fields are now showing the red border 

https://jira.rax.io/browse/JUPITER-6.   //Validation msg not showing now for email field on entering the input.

https://jira.rax.io/browse/JUPITER-11.  //Validation message "This username already exists. Please choose another one." is not disappearing on entering the Confirm Password

### CODE REVIEW

File: app/validators/index.js [hunk 1]
```diff
@@ -289,7 +289,12 @@ export const asyncValidate = (values, dispatch, { t = i18nT(), asyncErrors = {}
       throw _.merge(asyncErrors, error); // asyncErrors is undefined if none exist
     });
   }
-  return asyncValidatePassword(password, t).catch((error) => {
-    throw _.merge(asyncErrors, error);
-  });
+  return asyncValidatePassword(password, t).then((res) => {
```
Comment (abrande) [inline app/validators/index.js:292]:
```suggestion
  return asyncValidatePassword(password, t).then(() => {
```

**Review by abrande — APPROVED** [review]

File: app/validators/index.js (hunk 1 above)
Comment (anuritjain) [reply app/validators/index.js]: fixed

**Review by abrande — APPROVED** [review]

File: app/components/helix/inputTypes/Input.js [hunk 2]
```diff
@@ -17,7 +17,7 @@ const Input = (props) => {
   } = props;
   return (
     <div className={`${meta.asyncValidating ? 'async-validating' : ''} InputField`}>
-      <hx-text-control class={hxClassNames}>
+      <hx-text-control class={meta.touched && meta.invalid ? 'hxInvalid' : hxClassNames}>
```
Comment (nicko-winner) [inline app/components/helix/inputTypes/Input.js:20]: might be nice to have test coverage for this, if we refactor this to use a helix-react component the test will help ensure the component reactors as expected.

**Review by nicko-winner — APPROVED** [review]
Tested this is in staging, matches how helix does form validations for password validation. Blur triggers the first error, then any error after that is triggered by onChange. Nice solution! Good job 👍

File: app/components/helix/inputTypes/Input.js (hunk 2 above)
Comment (anuritjain) [reply app/components/helix/inputTypes/Input.js:20]: added test for it.

**Review by nicko-winner — APPROVED** [review]

**Review by vaibhavdubey08 — APPROVED** [review]

**Pull request merged**